matrix-synapse-riot/terraform/main.tf

```diff
@@ -0,0 +1,250 @@
+# Digital Ocean Tag for all server instances
+resource "digitalocean_tag" "matrix-home-service" {
+  name = "matrix-homeserver"
+}
+
+# Domain name managed by Digital Ocean
+#resource "digitalocean_domain" "matrix-home-service" {
+#  name        = "${file(var.domain_name)}"
+#  #ip_address  = "${digitalocean_droplet.matrix-server.ipv4_address}"
+#  #ip6_address = "${digitalocean_droplet.matrix-server.ipv6_address}"
+#}
+
+# Matrix server Droplet
+resource "digitalocean_droplet" "matrix-server" {
+  image              = "debian-9-x64"
+  name               = "matrix.${file(var.domain_name)}"
+  region             = "tor1"
+  size               = "1gb"
+  tags               = ["${digitalocean_tag.matrix-home-service.id}"]
+  private_networking = false
+  ipv6               = true
+  monitoring         = true
+  ssh_keys           = ["${file(var.ssh_fingerprint)}"]
+  connection {
+    user             = "root"
+    type             = "ssh"
+    private_key      = "${file(var.pvt_key)}"
+    timeout          = "2m"
+  }
+  provisioner "file" {
+    source           = "matrix-server"
+    destination      = "/tmp"
+  }
+  provisioner "remote-exec" {
+    inline           = [
+      "chmod +x /tmp/matrix-server/*.sh",
+      "/tmp/matrix-server/bootstrap.sh ${file(var.domain_name)}",
+    ]
+  }
+}
+
+# DNS records for Matrix
+resource "digitalocean_record" "matrix" {
+  domain = "${file(var.domain_name)}"
+  type   = "A"
+  name   = "matrix"
+  value  = "${digitalocean_droplet.matrix-server.ipv4_address}"
+  ttl    = "86400"
+}
+resource "digitalocean_record" "matrix-v6" {
+  domain = "${file(var.domain_name)}"
+  type   = "AAAA"
+  name   = "matrix"
+  value  = "${digitalocean_droplet.matrix-server.ipv6_address}"
+  ttl    = "86400"
+}
+resource "digitalocean_record" "matrix-caa" {
+  domain = "${file(var.domain_name)}"
+  type   = "CAA"
```

Comment:
This is for issuing wildcard cert to cjdns address?

Reply:
That is to set the Certification Authority Authorization to Let's Encrypt so that browsers will not trust another cert from a different CA. It is added because sometimes dehydrated will fail saying it cannot lookup the CAA record with SERVFAIL error.